pages/api/profiles.js

```diff
@@ -1,5 +1,24 @@
-import data from "../../.cache/data.json";
+import fs from "fs";
+import path from "path";
 
-export default function handler(req, res) {
-  res.status(200).json(data)
+export default async function handler(req, res) {
+  const relativeDir = path.resolve("./public", "profiles");
+  const files = fs.readdirSync(relativeDir);
+
+  const profiles = await Promise.all(
```

Comment:
This is a bit fragile imo. If one Promise fails, `Promise.resolve` wont resolve successfully. I think a propper error-handling needs to be added here. Its discussable if with this, or a future PR.